src/main/java/api/training/requests/PostRequest.java

```diff
@@ -0,0 +1,40 @@
+package api.training.requests;
+
+import api.training.exceptions.Exceptions;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class PostRequest {
+
+	private final HttpPost httpPost;
+
+	public PostRequest() {
+		httpPost = new HttpPost();
+	}
+
+	public void setURI(String uri) {
+		try {
+			httpPost.setURI(new URI(uri));
+		} catch (URISyntaxException e) {
+			throw new Exceptions.SetURIException();
+		}
+	}
+
+	public void setParams(List<NameValuePair> params) {
+		try {
+			httpPost.setEntity(new UrlEncodedFormEntity(params));
+		} catch (UnsupportedEncodingException e) {
+			throw new Exceptions.SetEntityException();
```

Comment:
One comment, but relates for all exception handling, I highly recommend to pass original exception as parameter (it will be easier to understand from stacktrace, what was wrong):
`throw new Exception.SetEntityException(e);`